### PR TITLE
Implement asynchronous support for Create, Update, Delete, and FileExists operations

### PR DESCRIPTION
**Pull request by ingshak**
Added CreateAsync, UpdateAsync, DeleteAsync, and FileExistsAsync methods to support asynchronous operations.

### CODE REVIEW

File: NGitLab/Impl/FilesClient.cs [hunk 1]
```diff
@@ -55,6 +70,32 @@ public bool FileExists(string filePath, string @ref)
         }
     }
 
+    public Task<bool> FileExistsAsync(string filePath, string @ref, CancellationToken cancellationToken = default)
+    {
+        return _api.Head().ExecuteAsync(_repoPath + $"/files/{EncodeFilePath(filePath)}?ref={@ref}", cancellationToken)
+            .ContinueWith(t =>
```
Comment (louis-z) [inline NGitLab/Impl/FilesClient.cs]: Please use `async`/`await`, as it make the code simpler and clearer.

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Impl/FilesClient.cs (hunk 1 above)
Comment (ingshak) [reply NGitLab/Impl/FilesClient.cs]: Done. Thanks for taking a look!

File: NGitLab/Impl/FilesClient.cs [hunk 2]
```diff
@@ -55,6 +70,25 @@ public bool FileExists(string filePath, string @ref)
...
+        catch (Exception ex) when (ex is OperationCanceledException ||
+                                  (ex is GitLabException gitLabException && gitLabException.StatusCode == HttpStatusCode.NotFound))
+
+        {
+            return false;
+        }
+        catch (Exception)
+        {
+            throw;
+        }
```
Comment (louis-z) [inline NGitLab/Impl/FilesClient.cs]: I think you can do the same as in the synchronous method:

```suggestion
        catch (GitLabException e) when (e.StatusCode == HttpStatusCode.NotFound)
        {
            return false;
        }
```

**Review by louis-z — CHANGES_REQUESTED** [review]

File: NGitLab/Impl/FilesClient.cs (hunk 2 above)
Comment (ingshak) [reply NGitLab/Impl/FilesClient.cs]: That's fair. Updated! Let me know.

**Pull request merged**